src/Shared/AzureManaged/DurableTaskSchedulerOptions.cs

```diff
@@ -0,0 +1,164 @@
+    public string EndpointAddress { get; set; } = string.Empty;
+
+    /// <summary>
+    /// Gets or sets the name of the task hub resource associated with the Durable Task Scheduler resource.
+    /// </summary>
+    [Required(ErrorMessage = "Task hub name is required")]
+    public string TaskHubName { get; set; } = string.Empty;
+
+    /// <summary>
+    /// Gets or sets the credential used to authenticate with the Durable Task Scheduler task hub resource.
+    /// </summary>
+    public TokenCredential? Credential { get; set; }
+
+    /// <summary>
+    /// Gets or sets the resource ID of the Durable Task Scheduler resource.
+    /// The default value is https://durabletask.io.
+    /// </summary>
+    public string ResourceId { get; set; } = "https://durabletask.io";
+
+    /// <summary>
+    /// Gets or sets the worker ID used to identify the worker instance.
+    /// The default value is a string containing the machine name, process ID, and a unique identifier.
+    /// </summary>
+    public string WorkerId { get; set; } = $"{Environment.MachineName},{Environment.ProcessId},{Guid.NewGuid():N}";
+
+    /// <summary>
+    /// Gets or sets a value indicating whether to allow insecure channel credentials.
+    /// This should only be set to true in development/testing scenarios.
+    /// </summary>
+    public bool AllowInsecureCredentials { get; set; }
+
+    /// <summary>
+    /// Creates a new instance of <see cref="DurableTaskSchedulerOptions"/> from a connection string.
+    /// </summary>
+    /// <param name="connectionString">The connection string to parse.</param>
+    /// <returns>A new instance of <see cref="DurableTaskSchedulerOptions"/>.</returns>
+    public static DurableTaskSchedulerOptions FromConnectionString(string connectionString)
+    {
+        return FromConnectionString(new DurableTaskSchedulerConnectionString(connectionString));
+    }
+
+    /// <summary>
+    /// Creates a new instance of <see cref="DurableTaskSchedulerOptions"/> from a parsed connection string.
+    /// </summary>
+    /// <param name="connectionString">The connection string to parse.</param>
+    /// <returns>A new instance of <see cref="DurableTaskSchedulerOptions"/>.</returns>
+    public static DurableTaskSchedulerOptions FromConnectionString(
+        DurableTaskSchedulerConnectionString connectionString) => new()
+        {
+            EndpointAddress = connectionString.Endpoint,
+            TaskHubName = connectionString.TaskHubName,
+            Credential = GetCredentialFromConnectionString(connectionString),
+        };
+
+    /// <summary>
+    /// Creates a gRPC channel for communicating with the Durable Task Scheduler service.
+    /// </summary>
+    /// <returns>A configured <see cref="GrpcChannel"/> instance that can be used to make gRPC calls.</returns>
+    public GrpcChannel CreateChannel()
+    {
+        Check.NotNullOrEmpty(this.EndpointAddress, nameof(this.EndpointAddress));
```

Comment:
`Check` is used for argument validation. Since these are not explicitly supplied arguments, use `Verify.NotNull` instead (you can also add a `Verify.NotNullOrEmpty` method to `Verify.cs` if you want.) Verify is used for verifying your expected state is correct.